scapy/layers/ipsec.py

```diff
@@ -127,8 +129,47 @@ class ESP(Packet):
 
 bind_layers(IP, ESP, proto=socket.IPPROTO_ESP)
 bind_layers(IPv6, ESP, nh=socket.IPPROTO_ESP)
-bind_layers(UDP, ESP, dport=4500)  # NAT-Traversal encapsulation
-bind_layers(UDP, ESP, sport=4500)  # NAT-Traversal encapsulation
+
+
+class NAT_T(Packet):
+    """
+    UDP Encapsulation of IPsec ESP Packets [RFC3948] (for NAT-Traversal)
+    """
+    name = 'NAT_T'
+
+    __slots__ = [
+        "_non_esp"
+    ]
+
+    fields_desc = [
+        ConditionalField(XIntField("non_esp", 0x0), lambda pkt: pkt._non_esp)
```

Comment:
Set the default to 1 too to match RFC?
(same as SPI)

Reply:
That would not make sense: The whole point about the non_esp field that it is either absent or consists of four zero bytes (which is an impossible value for an ESP spi).

In other words: the first four bytes of the UDP payload will be

- consumed by the `NAT_T` layer as `non_esp`  marker (and followed by the `init_cookie/init_SPI` of the `ISAKMP/IKEv2` layer), if all four bytes are zero
- consumbed by the `ESP` layer as `spi`, otherwise.

Reply:
See [RFC 3948, section 2.2](https://datatracker.ietf.org/doc/html/rfc3948#section-2.2)